gbpservice/nfp/orchestrator/modules/service_orchestrator.py

```diff
@@ -930,6 +972,14 @@ def handle_config_applied(self, event):
                  {'network_function_id':
                   network_function_id})
 
+    def handle_update_user_config_failed(self, event):
+        event_data = event.data
+        network_function_id = event_data['network_function_id']
+        LOG.error(_LE("NSO: updating user config failed, moving "
+                      "network function %(network_function_id)s to ERROR"),
+                  {'network_function_id': network_function_id})
+        self.handle_user_config_failed(event)
```

Comment:
In this method, the heat_stack_id is updated with event.data['heat_stack_id']
What if that value is None. I see its possible in line 767,
        request_data['heat_stack_id'] = self.config_driver.update_config(            network_function_details, stack_id)

Will delete node have problem in deleting the stack ?